tokio/src/sync/rwlock.rs

```diff
@@ -107,6 +107,9 @@ impl<'a, T> RwLockWriteGuard<'a, T> {
     /// Atomically downgrades a write lock into a read lock without allowing
     /// any writers to take exclusive access of the lock in the meantime.
     ///
+    /// **Note:** this won't *necessarily* allow any additional readers to acquire
+    /// locks, since [`RwLock`] is fair and it's possible a writer is next in line.
```

Comment:
nit:
```suggestion
    /// **Note:** This won't *necessarily* allow any additional readers to acquire
    /// locks, since [`RwLock`] is fair and it is possible that a writer is next
    /// in line.
```